Arrowgene.Ddon.Shared/Asset/QuestDropItemAsset.cs

```diff
@@ -0,0 +1,81 @@
+using Arrowgene.Ddon.Shared.AssetReader;
+using Arrowgene.Logging;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Arrowgene.Ddon.Shared.Asset
+{
+    public class QuestDropItemAsset
+    {
+        private static readonly ILogger Logger = LogProvider.Logger(typeof(QuestDropItemAsset));
+
+        public QuestDropItemAsset()
+        {
+
+            QuestDropsTables = new();
+            EnemyLevelMap = new();
+        }
+        // <dropTableId, DropTable>>
+        private Dictionary<uint, DropsTable> QuestDropsTables { get; set; }
+
+        // <enemyId, <enemyLevel, DropTableId>>
+        private Dictionary<uint, SortedList<ushort, uint>> EnemyLevelMap { get; set; }
+
+        // This is to be used within the quest drop table serializer for adding the default drop tables
+        public void AddDropTable(uint enemyId, ushort levelMin, ushort levelMax, DropsTable dropTable)
+        {
+            if (!EnemyLevelMap.ContainsKey(enemyId))
+            {
+                EnemyLevelMap[enemyId] = new SortedList<ushort, uint>();
+            }
+
+            // Create a mapping of the enemy id level values to the correct drop table id.
+
+            for (ushort i = levelMin; i <= levelMax; i++)
+            {
+                EnemyLevelMap[enemyId].Add(i, dropTable.Id);
+            }
+
+            // Add the drop table with the drop table id to the main DropsTable
+            QuestDropsTables[dropTable.Id] = dropTable;
+        }
+
+        public void AddDropTable(uint enemyId, ushort levelMin, DropsTable dropTable)
+        {
+            if (!EnemyLevelMap.ContainsKey(enemyId))
+            {
+                EnemyLevelMap[enemyId] = new SortedList<ushort, uint>();
+            }
+
+            // Create a mapping of the enemy id level values to the correct drop table id.
+
+            EnemyLevelMap[enemyId].Add(levelMin, dropTable.Id);
```

Comment:
This is one of the quirks we're working with. So some enemies don't drop things below a level.

For example: A < lvl 50 Rabbit may not drop any loot at all, but above that it does so that's where this sort of thing happens. If an enemy is above level 50 it uses the highest level number in the SortedList to apply those drops, but under that nothing appears.

This is all populated when the server is starting up and reading files, so if there is overlap then the app will scream and error and won't run until it's fixed. I tested this by intentionally causing overlaps and watched it fall over 😄

I hope that clears it up